index.js

```diff
@@ -31,13 +31,85 @@ SOFTWARE.
 var parseXML     = require('xml2js').parseString
   , url          = require('url')
   , EventEmitter = require('events').EventEmitter
+  , HttpProxyAgent = require('http-proxy-agent')
+  , HttpsProxyAgent = require('https-proxy-agent')
   ;
 
 // These numbers were obtained by measuring and averaging both using this module and the official speedtest.net
 var speedTestDownloadCorrectionFactor = 1.135
   , speedTestUploadCorrectionFactor   = 1.139
   ;
 
+// 	Proxy by parameter
+var proxyOptions = null;
+
+// Search a proxy (http|https) in env with case-insensitive
+var proxyHttpEnv = findPropertiesInEnvInsensitive("HTTP_PROXY");
+var proxyHttpsEnv = findPropertiesInEnvInsensitive("HTTPS_PROXY");
+
+function findPropertiesInEnvInsensitive(prop) {
+    prop = (prop + "").toLowerCase();
+    for(var p in process.env){
+        if(process.env.hasOwnProperty(p) && prop == (p+ "").toLowerCase()){
+            return process.env[p];
+        }
+    }
+    return null;
+}
+
+//set the proxy agent for each http request
+// priority : 
+// 1 - proxyOptions
+// 2 - proxyHttpEnv (HTTP_PROXY)
+// 3 - proxyHttpsEnv (HTTPS_PROXY)
+function proxy(options) {
+    if (proxyHttpEnv == null && proxyHttpsEnv == null && proxyOptions == null)
+    {
+        return;
+    }
+    var proxy = null;
+    var isSSL = false;
+    var haveHttp = false;
+	// Test the proxy parameter first for priority
+    if (proxyOptions != null)
+    {
+        if (proxyOptions.startsWith("https:")) {
+            isSSL = true
+        }
+        proxy = proxyOptions;
+    }
+    else {
+	// Test proxy by env
+        proxy = proxyHttpEnv;
+		if (proxyHttpEnv != null) {
+			//for support https in HTTP_PROXY env var
+			if (proxyHttpEnv.startsWith("https:"))
+			{
+				isSSL = true
+			} else {
+				haveHttp = true;
+			}
+
+		} else if (proxyHttpsEnv != null) {
+			isSSL = true
+			proxy = proxyHttpsEnv;
+		}
+        // for http priority
+        if (proxyHttpEnv != null && !proxyHttpEnv.startsWith("https:")) {
```

Comment:
Should this include the following line?
```js
  proxy = proxyHttpEnv;
```

Also, this should be properly indented with 2 spaces for each level.
There is no need to test for `null`, just use `if (proxyHttpEnv && !proxyHttpEnv.startsWith('https:')) {`.

Reply:
proxy = proxyHttpEnv; is declare by default at the beginning of the block, and when it is necessary I set the https with proxy = proxyHttpsEnv;

Reply:
done